test/test_datasets.py

```diff
@@ -2205,5 +2205,45 @@ def inject_fake_data(self, tmpdir: str, config):
         return len(sampled_classes * n_samples_per_class)
 
 
+class SUN397TestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.SUN397
+    FEATURE_TYPES = (PIL.Image.Image, int)
+
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        split=("train", "test"),
+        partition=(1, 10, None),
+    )
+
+    def inject_fake_data(self, tmpdir: str, config):
+        data_dir = pathlib.Path(tmpdir) / "SUN397"
+
+        data_dir.mkdir(parents=True)
+
+        num_images_per_class = 5
+        sampled_classes = ("abbey", "airplane_cabin", "airport_terminal")
+        im_paths = []
+        for cls in sampled_classes:
+            image_folder = data_dir / cls[0]
+            im_paths.append(
```

Comment:
By `extend`'ing here, you can avoid the list of lists structure.

```suggestion
            im_paths.extend(
```